polaris-service/src/main/java/org/apache/polaris/service/context/DefaultContextResolver.java

```diff
@@ -44,12 +44,11 @@
  * <p>Example: principal:data-engineer;password:test;realm:acct123
  */
 @JsonTypeName("default")
-public class DefaultContextResolver
-    implements RealmContextResolver, CallContextResolver, ConfigurationStoreAware {
+public class DefaultContextResolver extends RealmContextResolver
+    implements CallContextResolver, ConfigurationStoreAware {
```

Comment:
I believe it's unnecessary to rely on inheritance to access methods from RealmContextResolver. Instead, we can use composition, which I think is more appropriate in this case.

We can start by making the resolver a singleton, and then utilize it as a class field within this context. This approach simplifies the design and avoids the potential pitfalls of an inheritance-based model.

Let me know your thoughts.

Reply:
It's not _necessary_ to rely on inheritance, but DefaultContextResolver _is_ a RealmContextResolver. I don't think composition is more appropriate given the semantic relationship between these types.

Reply:
Looked a bit more, composition isn't a good way.  But I also think it's unnecessary to combine two interfaces in one solid class, it's not a blocker for this PR thought. We can fix it later.